### PR TITLE
[terra-clinical-detail-view] Use semantic html elements for accessibility

### PR DESCRIPTION
**Pull request by kolkheang**
### Summary

**What was changed:**
Update DetailView in terra-clinical-detail-view to use semantic HTML elements, specifically:
- the subtitle is now using `<p>` element instead of `<div>`
- the footer is now using `<p>` with role=note aria instead of `<div>`
- the whole detail view is now within a `<section>` instead of `<div>`
- the graph element is now wrap within a `<figure>`

The rest of the changes such as using proper heading level for title, secondary title, detail list heading, and definition list for the detail list will be part of other PRs.

**Why it was changed:**
To provide better accessibility support to screen readers for the changes listed above

### Testing

This change was tested using:

- [x] WDIO
- [x] Jest
- [x] Visual testing (please attach a screenshot or recording)
- [ ] Other (please describe below)
- [ ] No tests are needed

### Reviews

In addition to engineering reviews, this PR needs:

- [x] UX review
- [x] Accessibility review
- [ ] Functional review

### Additional Details

**This PR resolves:**

UXPLATFORM-9075 

---

Thank you for contributing to Terra.
@cerner/terra

### CODE REVIEW

File: packages/terra-clinical-detail-view/src/DetailView.jsx [hunk 1]
```diff
@@ -98,10 +98,11 @@ const DetailView = (props) => {
   ));
   const subtitleElements = subtitles.map((subTitle, i) => (
     // eslint-disable-next-line react/no-array-index-key
-    <div className={cx('subtitle')} key={`${i}`}>{subTitle}</div>
+    <p className={cx('subtitle')} key={`${i}`}>{subTitle}</p>
```
[the remark this answers is not in the document]
Comment (kolkheang) [reply packages/terra-clinical-detail-view/src/DetailView.jsx:101]: No, I don't think so. role="note" was called out to be updated, and specific to the `footer` message at the bottom of the component.

**Comment by kolkheang** [comment]
There should not be any visual changes to tests/wdio. Using the swipe method to compare the deleted vs added images, there aren't any visual changes either.

File: packages/terra-clinical-detail-view/src/DetailView.jsx [hunk 2]
```diff
@@ -118,19 +119,19 @@ const DetailView = (props) => {
...
   return (
-    <div {...attributes}>
+    <section {...attributes}>
       <div className={cx('titles-section', { 'titles-smaller': isSmallerTitles })}>
         {titleElement}
         {secondaryTitlesElements}
         {subtitleElements}
         {accessoryElement}
       </div>
       {graph && divider}
```
Comment (zxeleven) [inline packages/terra-clinical-detail-view/src/DetailView.jsx:129]: nit: should this be `graphElement` as well?

File: packages/terra-clinical-detail-view/CHANGELOG.md [hunk 3]
```diff
@@ -2,6 +2,9 @@
 
 ## Unreleased
 
+* Changed
+  * Update DetailView component to use semantic html elements
```
Comment (chenyixuan625) [inline packages/terra-clinical-detail-view/CHANGELOG.md:6]: nit: Missing the period in the end.

**Comment by chrismichalewicz** [comment]
@kolkheang  Is there a good example I could use to test for JAWS? The examples I find through the deployment link I believe are reading out correctly but it's hard to follow as it is pretty much a list of "A Detail List Item" repeated so it's hard to follow along with the structure.

**Comment by kolkheang** [comment]
> @kolkheang Is there a good example I could use to test for JAWS? The examples I find through the deployment link I believe are reading out correctly but it's hard to follow as it is pretty much a list of "A Detail List Item" repeated so it's hard to follow along with the structure.

Apologies for not listing the specific changes within this PR. I've updated the things that are part of this PR.
```
What was changed:
Update DetailView in terra-clinical-detail-view to use semantic HTML elements, specifically:

    the subtitle is now using

    element instead of
    the footer is now using

    with role=note aria instead of
    the whole detail view is now within a
    instead of
    the graph element is now wrap within a

The rest of the changes such as using proper heading level for title, secondary title, detail list heading, and definition list for the detail list will be part of other PRs.
```

With the latest commit, I'm including the `graph` prop in the last example. It's using <p> text right now. I'm open to suggestion on what would be a good example for the `graph` prop.

**Comment by chrismichalewicz** [comment]
@kolkheang  I updated the a11y reviewed label. It would still be good to have an engineer from terra on this as well to ensure there are no concerns with the technical aspects. @JessieRandle can help with identifying someone to look at this.  Thanks!

**Review by zxeleven — APPROVED** [review]

**Review by chenyixuan625 — APPROVED** [review]

**Review by sdadn — APPROVED** [review]

**Review by sycombs — APPROVED** [review]

**Comment by sdadn** [comment]
Since this updates the tags and html content of the component, this will need UX review as well. But it looks good from an engineering standpoint.

File: packages/terra-clinical-detail-view/src/DetailView.jsx [hunk 4]
```diff
@@ -118,19 +121,19 @@ const DetailView = (props) => {
   }
 
   return (
-    <div {...attributes}>
+    <section {...attributes}>
       <div className={cx('titles-section', { 'titles-smaller': isSmallerTitles })}>
         {titleElement}
```
Comment (brooksienoodlesoup) [inline packages/terra-clinical-detail-view/src/DetailView.jsx:126]: This section element should have an aria-labelledby pointing to an id value that is on the heading.  section elements aren't announced unless they have an accessible name associated.  Using aria-labelledby in this way creates the accessible name for the section content.

**Review by brooksienoodlesoup — COMMENTED** [review]
Changing div to p should not create a noticeable change in the announcement of the content, so not hearing a difference is the expected behavior.  Announcement of figure and figcaption content does not include the announcement of the word "figure" or similar language. So not hearing figure or figcaption is expected behavior.

File: packages/terra-clinical-detail-view/src/DetailView.jsx (hunk 4 above)
Comment (kolkheang) [reply packages/terra-clinical-detail-view/src/DetailView.jsx:126]: The `title` prop is optional. We may not be able to tie this to the id in the title if it's not passed in. We also need to include id in this PR https://github.com/cerner/terra-clinical/pull/858/files#diff-890b5219922c74bbe07ef699395ae882c595463b51f77a947d6c0fe5ad8e7e0bR36

Comment (scottwilmarth) [reply packages/terra-clinical-detail-view/src/DetailView.jsx:126]: As noted previously, we'll add the title prop to a new Jira.

**Review by scottwilmarth — APPROVED** [review]
Reviewed and approved.

**Review by sdadn — APPROVED** [review]

**Review by sycombs — APPROVED** [review]

**Comment by sdadn** [comment]
Looks like there are some merge conflicts; can you resolve those before this can be merges?

**Comment by kolkheang** [comment]
@sycombs  / @sdadn could you take another look and merge, please?
